plugins/speccy.py

```diff
@@ -0,0 +1,134 @@
+
+    body = soup.body
+
+    data = []
+
+    os_spec = body.find("div", text='Operating System')
+    if os_spec is not None:
+        data.append(
+            "\x02OS:\x02" + " " + os_spec.next_sibling.next_sibling.text)
+
+    def cspace():
+        c_space_spec = body.find("div", class_="datavalue", text='C:')
+        if c_space_spec is not None:
+            c_space_spec_found = c_space_spec.parent.parent
+            c_space_data = c_space_spec_found.find("div", text='Free Space:\xA0', class_="datakey").parent
+            return c_space_data.find(class_="datavalue").text
+
+    c_space_spec = cspace()
+    if c_space_spec is not None:
+        data.append("\x02Space Left (C:\):\x02" + " " + c_space_spec)
+
+    ram_spec = body.find("div", text='RAM')
+    if ram_spec is not None:
+        data.append(
+            "\x02RAM:\x02" + " " + ram_spec.next_sibling.next_sibling.text)
+
+    def ramusg():
+        ram_usg_spec = body.find("div", class_="blue clear", text='Physical ramory')
+        if ram_usg_spec is not None:
+            ram_usg_spec_found = ram_usg_spec.next_sibling.next_sibling
+            ram_usg_data = ram_usg_spec_found.find("div", text='Memory Usage:\xA0', class_="datakey").parent
+            return ram_usg_data.find(class_="datavalue").text
+
+    ram_usg_spec = ramusg()
+    if ram_usg_spec is not None:
+        data.append("\x02RAM USG:\x02" + " " + ram_usg_spec)
+
+    cpu_spec = body.find("div", text='CPU')
+    if cpu_spec is not None:
+        data.append(
+            "\x02CPU:\x02" + " " + cpu_spec.next_sibling.next_sibling.text)
+
+    def gpuspec():
+        gpu_find = body.find("div", text='Graphics')
+        if gpu_find is not None:
+            gpu_find = gpu_find.next_sibling.next_sibling.text
+            gpu_spec = ""
+            for gpu_string in GPU_RE.finditer(gpu_find):
+                gpu_spec += gpu_string.group()
+                return gpu_spec
+
+    gpu_spec = gpuspec()
+    if gpu_spec is not None:
+        data.append("\x02GPU:\x02" + " " + gpu_spec)
+
+    has_badware = False
+
+    pico_spec = body.find("div", text=PICO_RE)
+    if pico_spec is not None:
+        has_badware = True
+        data.append("\x02Badware:\x02" + " " + pico_spec.text)
+
+    kms_spec = body.find("div", text=KMS_RE)
+    if kms_spec is not None:
+        has_badware = True
+        data.append("\x02Badware:\x02" + " " + kms_spec.text)
+
+    booster_spec = body.find("div", text=BOOSTER_RE)
+    if booster_spec is not None:
+        has_badware = True
+        data.append("\x02Badware:\x02" + " " + booster_spec.text)
+
+    reviver_spec = body.find("div", text=REVIVER_RE)
+    if reviver_spec is not None:
+        has_badware = True
+        data.append("\x02Badware:\x02" + " " + reviver_spec.text)
+
+    killer_spec = body.find("div", text=KILLER_RE)
+    if killer_spec is not None:
+        has_badware = True
+        data.append("\x02Badware:\x02" + " " + killer_spec.text)
```

Comment:
All of the badware logic could probably be collapsed to something like
```python
badware_patterns = [KILLER_RE, REVIVER_RE, BOOSTER_RE, KMS_RE, PICO_RE]
badware = body.find_all("div", text=badware_patterns)
if badware:
    for tag in badware:
        data.append("\x02Badware:\x02" + " " + tag.text)
else:
    data.append("\x02No Badware\x02")
```
Rather than duplicating the code. Then any new Badware patterns just need to be added to that list, and that list could even be defined with the rest of the regexs like:
```python
BADWARE_PATTERNS = [...]
```

Reply:
done, lots better